example/particles/particles.cpp

```diff
@@ -259,47 +339,82 @@ TaskStatus TransportParticles(MeshBlock *pmb, StagedIntegrator *integrator) {
 
   auto swarm_d = swarm->GetDeviceContext();
 
-  ParArrayND<Real> t("time", max_active_index + 1);
-
-  // Simple particle push: push particles half a zone width until they have
-  // traveled one integrator timestep's worth of time
-  pmb->par_for(
-      "TransportParticles", 0, max_active_index, KOKKOS_LAMBDA(const int n) {
-        if (swarm_d.IsActive(n)) {
-          t(n) = 0.;
-          Real v = sqrt(vx(n) * vx(n) + vy(n) * vy(n) + vz(n) * vz(n));
-          while (t(n) < dt) {
-            Real dt_cell = dx_push / v;
-            Real dt_end = dt - t(n);
-            Real dt_push = std::min<Real>(dt_cell, dt_end);
-
-            x(n) += vx(n) * dt_push;
-            y(n) += vy(n) * dt_push;
-            z(n) += vz(n) * dt_push;
-            t(n) += dt_push;
-
-            // Periodic boundaries
-            if (x(n) < x_min) {
-              x(n) = x_max - (x_min - x(n));
+  // Simple particle push: push particles half the smallest zone width until they have
+  // traveled one integrator timestep's worth of time. Particles orbit the origin.
+  if (orbiting_particles) {
+    // Particles orbit the origin
+    pmb->par_for(
+        "TransportOrbitingParticles", 0, max_active_index, KOKKOS_LAMBDA(const int n) {
+          if (swarm_d.IsActive(n)) {
+            Real v = sqrt(vx(n) * vx(n) + vy(n) * vy(n) + vz(n) * vz(n));
+            while (t(n) < t0 + dt) {
+              Real dt_cell = dx_push / v;
+              Real dt_end = t0 + dt - t(n);
+              Real dt_push = std::min<Real>(dt_cell, dt_end);
+
+              Real r = sqrt(x(n) * x(n) + y(n) * y(n) + z(n) * z(n));
+
+              x(n) += vx(n) * dt_push;
+              y(n) += vy(n) * dt_push;
+              z(n) += vz(n) * dt_push;
+              t(n) += dt_push;
+
+              // Force point back onto spherical shell
+              Real r_tmp = sqrt(x(n) * x(n) + y(n) * y(n) + z(n) * z(n));
+              x(n) *= r / r_tmp;
+              y(n) *= r / r_tmp;
+              z(n) *= r / r_tmp;
```

Comment:
Yes I'm doing this to ensure the example behaves itself for any runtime. We don't provide any integration methods in parthenon itself so I don't know that it's worth using this example to test a method written just for this example. Because of that I was wondering whether it's worth it to include this, but I wrote this orbiting particles code out before wondering that, so I committed it. Do you think we should still keep this?